### ISSUE TITLE
Bad SSL logic for Erlang 18.0 and hackney 1.6.3

### ISSUE THREAD

**Issue by loguntsov** [benoitc/hackney#358]
We uses hackney 1.6.1 with Erlang 18.0. 
Everything works fine with HTTPs.

When i updated hackney for 1.6.3, i got this error:

```
,{{function_clause,[{public_key,ec_key,[notsup,{namedCurve,{1,2,840,10045,3,1,7}}],[{file,"public_key.erl"},{line,1001}]},{ssl_connection,calculate_secret,3,[{file,"ssl_connection.erl"},{line,1526}]},{tls_connection,next_state,4,[{file,"tls_connection.erl"},{line,466}]},{gen_fsm,handle_msg,7,[{file,"gen_fsm.erl"},
```

I tried reproduce it for my local Erlang (18.3):

```
begemot@big-comp:~$ erl
Erlang/OTP 18 [erts-7.3] [source-d2a6d81] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V7.3  (abort with ^G)
1> application:ensure_all_started(crypto).
3> public_key:generate_key({namedCurve,{1,2,840,10045,3,1,7}}).
{'ECPrivateKey',1,
                <<9,138,150,128,186,175,90,216,66,103,86,170,3,152,203,53,
                  71,142,153,6,30,125,130,66,170,200,...>>,
                {namedCurve,{1,2,840,10045,3,1,7}},
                <<4,137,146,92,144,2,31,77,228,91,66,233,250,22,253,124,
                  18,139,180,56,64,37,137,206,...>>}
4> 
```

And my Erlang for development (18.0):

```
[root@dev-master1 ejabberd]# erl
Erlang/OTP 18 [erts-7.0] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V7.0  (abort with ^G)
1> 
1> 
1> application:ensure_all_started(crypto).
{ok,[crypto]}
2> public_key:generate_key({namedCurve,{1,2,840,10045,3,1,7}}).
** exception error: no function clause matching public_key:ec_key(notsup,{namedCurve,{1,2,840,10045,3,1,7}}) (public_key.erl, line 1001)
3> 
```

So see the main issue was fixed in Erlang. 
But right now i expects that requirements for hackney 1.6.3 will be like Erlang 18.3 or higher.

So could you help me localize this problem and to help make hackney and Erlang more better.
Thanks.

P.S. What info do you need additionally ?

**Comment by benoitc** [comment]
The issue is the new ciphers requirements setting. I will remove it for the 1.6.x branch.

**Comment by benoitc** [comment]
this is fixed in 4f86b6a91d32f1d7e890d161e92e533b332a545. so machines that have not openssl built with ecc will still be able to connect. A warning will be displayed. Assuming this is what you really want ;)
